doc/glossary.rst

```diff
@@ -0,0 +1,973 @@
+        TODO
+
+    ``get_params``
+        TODO
+
+    ``fit_predict``
+        TODO
+
+        See :term:`labels_`
+
+    ``fit_transform``
+        TODO
+
+        Syntactic sugar.
+        Efficiency.
+        ``transform`` may not be available.
+        ``fit_transform`` can be different, as in stacking.
+
+        Ordinarily should not be applied to the entirety of a dataset, only the
+        training data, to avoid :term:`leakage`.
+
+    ``fit``
+        The ``fit`` method is provided on every estimator. It usually takes some
+        :term:`samples` ``X``, :term:`targets` ``y`` if the model is supervised,
+        and potentially other :term:`sample properties` such as
+        :term:`sample_weight`.  It should:
+
+        * clear any prior :term:`attributes` stored on the estimator, unless
+          :term:`warm_start` is used;
+        * validate and interpret any :term:`parameters`, ideally raising an
+          error if invalid;
+        * validate the input data;
+        * estimate and store model attributes from the estimated parameters and
+          provided data; and
+        * return the now :term:`fitted` estimator to facilitate method
+          chaining.
+
+        :ref:`glossary_target_types` describes possible formats for ``y``.
+
+    ``partial_fit``
+        Facilitates fitting an estimator in an online fashion.  Unlike ``fit``,
+        repeatedly calling ``partial_fit`` does not clear the model, but
+        updates it with respect to the data provided. The portion of data
+        provided to ``partial_fit`` may be called a mini-batch.
+        Each mini-batch must be of consistent shape, etc.
+
+        ``partial_fit`` may also be used for :term:`out-of-core` learning,
+        although limited to the case where learning can be performed online,
+        i.e. the model is usable after each ``partial_fit`` and there is no
+        separate processing needed to finalize the model.
+        :class:`cluster.Birch` introduces the convention that calling
+        ``partial_fit(X)`` will produce a model that is not finalized, but the
+        model can be finalized by calling ``partial_fit()`` i.e. without
+        passing a further mini-batch.
+
+        Generally, estimator parameters should not be modified between calls
+        to ``partial_fit``, although ``partial_fit`` should validate them
+        as well as the new mini-batch of data.
+
+        Like ``fit``, ``partial_fit`` should return the estimator object.
+
+        To clear the model, a new estimator should be constructed, for instance
+        with :func:`base.clone`.
+
+    ``predict``
+        Makes a prediction for each sample. In a :term:`classifier` or
+        :term:`regressor`, this prediction is in the same target space used in
+        fitting (e.g. one of {'red', 'amber', 'green'} if the `y` in fitting
+        consisted of these strings).  In a :term:`clusterer` or :term:`outlier
+        detector` the prediction is an integer.
```

Comment:
mention -1 for outlier detection and DBSCAN

Reply:
mentioned below.